### ISSUE TITLE
Support `@var` annotations for promoted properties

### ISSUE THREAD

**Issue by dantleech** [phpactor/phpactor#2417]
![image](https://github.com/phpactor/phpactor/assets/530801/345aa145-6123-4fcb-b6d3-f74f5a234b5f)

**Comment by CalebDW** [comment]
I don't think this makes sense...promoted properties should be typed via the `@param` tag above the constructor

From your example:
```php
/** @param list<\PhpTui\Tui\Model\Widget> $widgets */
public function __construct(
    public array $widgets
) {}

**Comment by CalebDW** [comment]
Hmm...looks like PHP supports it though so just a matter of personal preference:

![image](https://github.com/phpactor/phpactor/assets/4176520/a32337f0-07e3-4957-be54-25422c873b05)

**Comment by dantleech** [comment]
PHPStan supports this too. For me it makes sense, it saves repeasting the var name :)
